documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+ 
+                leaf ip_version {
+                    type string {
+                        pattern "IPv4|IPv6"
+                    }
+                }
+
+                leaf routing_type {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ROUTING_TYPE/dash:DASH_ROUTING_TYPE_LIST/dash:name;
+                    }
+                }
+
+                leaf vnet {
+                    when "((current()/../routing_type = 'vnet') or (current()/../routing_type = 'vnet-direct'))";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_VNET/dash:DASH_VNET_LIST/dash:name;
+                    }
+                }
+
+                leaf customer_addr {
+                    when "((current()/../routing_type = 'vnet') or (current()/../routing_type = 'vnet-direct'))";
+                    description "if routing_type is vnet-direct"
+                    type inet:ip-address;
+                }
+
+                leaf appliance {
+                    when "(current()/../routing_type = 'appliance')";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_APPLIANCE/dash:DASH_APPLIANCE_LIST/dash:name;
+                    }
+                }
+
+                leaf overlay_ip {
+                    when "(current()/../routing_type = 'servicetunnel')";
+                    description "Overlay ip for service tunnel"
+                    type inet:ip-address;
+                }
+
+                leaf underlay_ip {
+                    when "(current()/../routing_type = 'servicetunnel')";
+                    description "Underlay ip for service tunnel"
+                    type inet:ip-address;
+                }
+
+                leaf overlay_sip {
+                    when "(current()/../routing_type = 'servicetunnel')";
+                    description "Overlay source ip for service tunnel"
+                    type inet:ip-address;
+                }
+
+                leaf overlay_dip {
+                    when "(current()/../routing_type = 'servicetunnel')";
+                    description "Overlay dst ip for service tunnel"
+                    type inet:ip-address;
+                }
+
+                leaf metering_bucket {
+                    type leafref {
+                        path TBD;
+                    }
+                }
+
+            } /* end of list DASH_ROUTE_TABLE_LIST */
+        } /* end of container DASH_ROUTE_TABLE */
+
+        container DASH_MAPPING_TABLE {
+            description "DASH Mapping table associated to a Vnet; part of APP_DB";            
+
+            list DASH_MAPPING_TABLE_LIST {
+                key "vnet ip_addr";
+
+                leaf vnet {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_VNET/dash:DASH_VNET_LIST/dash:name;
```

Comment:
Since DASH_VNET also in the same file...please use "path ../../../DASH_VNET/DASH_VNET_LIST/name", please address this in all places.